utam-core/src/main/java/utam/core/framework/element/NavigationImpl.java

```diff
@@ -28,9 +31,13 @@ public class NavigationImpl implements Navigation {
   static final String ERR_NO_WINDOW_WITH_URL = "can't find window with url %s";
   static final String ERR_NO_INITIAL_WINDOW =
           "window with url %s not found, and the previous window was closed";
+  static final String ERR_SETUP_NOT_RUN = "setupWaitForNewWindow must be run before waitForNewWindow";
 
   private final Driver driverAdapter;
   private final PageObjectsFactory factory;
+  private boolean setupForNewWindow = false;
+  private int oldWindowCount;
+  Set<String> oldWindowHandles;
```

Comment:
This seems like a fair amount of moving parts for this. Do we really need a `boolean` tracking variable and and `int` of the window count? Using `oldWindowHandles == null` for the former and `oldWindowHandles.size()` for the latter seems reasonable and readable, especially since these variables are only implementation details.